src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -829,6 +830,141 @@ ov::pass::NopSliceBeforeGatherElements::NopSliceBeforeGatherElements() {
+            !check_mask(strided_slice_node->get_new_axis_mask()) ||
+            !check_mask(strided_slice_node->get_ellipsis_mask())) {
+            return false;
+        }
+        // check that that we will take all values
+        if (node->get_input_size() == 4 && !op::util::is_constant_and_all_values_equal_int(node->input_value(3), 1)) {
+            return false;
+        }
+
+        auto align_vectors = [](std::vector<int64_t>& vec_1, std::vector<int64_t>& vec_2) {
+            auto max_size = std::max(vec_1.size(), vec_2.size());
+            while (vec_1.size() < max_size) {
+                vec_1.push_back(0);
+            }
+            while (vec_2.size() < max_size) {
+                vec_2.push_back(0);
+            }
+            return;
+        };
+        auto begin_node = strided_slice_node->get_input_node_shared_ptr(1);
+        if (const auto& begin_constant_node = ov::util::get_constant_from_source(begin_node)) {
+            auto values = begin_constant_node->cast_vector<int64_t>();
+            auto begin_mask = strided_slice_node->get_begin_mask();
+            // align begin_mask and values_vec by length
+            align_vectors(values, begin_mask);
+            for (size_t i = 0; i < begin_mask.size(); ++i) {
+                // if mask == 1 then ignore the begin_mask_value else check
+                // if values[i] == 0 then take whole tensor else take part of a tensor
+                if (!begin_mask[i] && values[i]) {
+                    return false;
+                }
+            }
+        } else {
+            return false;
+        }
+
+        auto end_node = strided_slice_node->get_input_node_shared_ptr(2);
+        if (const auto& end_constant_node = ov::util::get_constant_from_source(end_node)) {
+            auto values = end_constant_node->cast_vector<int64_t>();
+            auto end_mask = strided_slice_node->get_end_mask();
+            // align end_mask and values_vec by length
+            align_vectors(values, end_mask);
+            for (size_t i = 0; i < end_mask.size(); ++i) {
+                // if mask == 1 then ignore the begin_mask_value else check
+                // if values[i] == max then take whole tensor else take part of a tensor
+                if (!end_mask[i] && values[i] != std::numeric_limits<int64_t>::max()) {
+                    return false;
+                }
+            }
+        } else {
+            return false;
+        }
+        return replace_output_update_name(strided_slice_node->output(0), strided_slice_node->input_value(0));
+    };
+    auto m = std::make_shared<pattern::Matcher>(pattern, matcher_name);
+    register_matcher(m, matcher_pass_callback);
+}
+
+ov::pass::NopStridedSliceByShape::NopStridedSliceByShape() {
+    MATCHER_SCOPE(NopStridedSliceByShape);
+    auto slice = pattern::wrap_type<op::v8::Slice>();
+
+    auto input = pattern::any_input();
+    auto begin_const = pattern::any_input();
+    auto end_const = pattern::any_input();
```

Comment:
Can it be a case that begin_const and end_const are under ShapeOf sub-graph and StridedSlice generates output shape the same as input shape up to some values of `begin` and `end`. After some limit `begin` and `end` changes and output shape will be not equal to input shape. May be `begin` and `end` use `Maximum` and `Minimum` operation like sort of clipping?

Reply:
in this case, I believe StridedSlice will generate dynamic shape, but here we consider both (input, output) shapes are static. I think the shape inference function uses the begin/end values, not only shapes, to calculate the shape output.

Reply:
I think shape inference uses values of `begin` and `end` and these values can represent not continuous function on input shape so that output shape will be not equal input shape after calling reshape method of `ov::Model`.